nucypher/blockchain/eth/events.py

```diff
@@ -77,3 +80,44 @@ def __getattr__(self, event_name: str):
     def __iter__(self):
         for event_name in self.names:
             yield self[event_name]
+
+
+class ContractEventsThrottler:
+    """
+    Enables Contract events to be retrieved in batches.
+    """
+    # default to 1000 - smallest default heard about so far (alchemy)
+    DEFAULT_MAX_BLOCKS_PER_CALL = int(os.environ.get(NUCYPHER_EVENTS_THROTTLE_MAX_BLOCKS, 1000))
+
+    def __init__(self,
+                 agent: 'EthereumContractAgent',
+                 event_name: str,
+                 from_block: int,
+                 to_block: int = None,  # defaults to latest block
+                 max_blocks_per_call: int = DEFAULT_MAX_BLOCKS_PER_CALL,
+                 **argument_filters):
+        if not agent:
+            raise ValueError(f"Contract agent must be provided")
+        if not event_name:
+            raise ValueError(f"Event name must be provided")
+
+        self.event_filter = agent.events[event_name]
+        self.from_block = from_block
+        self.to_block = to_block if to_block else agent.blockchain.client.block_number
+        # validity check of block range
+        if to_block <= from_block:
+            raise ValueError(f"Invalid block range provided ({from_block} - {to_block})")
```

Comment:
What do you think about allowing `to_block='latest'`?

Reply:
`to_block` is currently provided as an int in the constructor, but if it is None (default) then the latest block is used - see L106. I didn't want `to_block` to be a string since it just causes more issues with validating the string.